service/src/main/java/bio/terra/workspace/service/workspace/CloudSyncRoleMapping.java

```diff
@@ -11,34 +12,52 @@
-  // be per-dataset.
+
+  // Note that custom roles defined at the project level cannot contain the
+  // "resourcemanager.projects.list" permission, even though it was previously included here.
+  // See https://cloud.google.com/iam/docs/understanding-custom-roles#known_limitations
   private static final List<String> READER_PERMISSIONS =
       ImmutableList.of(
-          "roles/bigquery.jobUser",
-          "roles/lifesciences.viewer",
-          "roles/serviceusage.serviceUsageViewer");
+          "bigquery.jobs.create",
+          "lifesciences.operations.get",
+          "lifesciences.operations.list",
+          "resourcemanager.projects.get",
+          "monitoring.timeSeries.list",
+          "serviceusage.operations.get",
+          "serviceusage.operations.list",
+          "serviceusage.quotas.get",
+          "serviceusage.services.get",
+          "serviceusage.services.list");
   private static final List<String> WRITER_PERMISSIONS =
       new ImmutableList.Builder<String>()
           .addAll(READER_PERMISSIONS)
           .add(
               // TODO(wchambers): Revise service account permissions when there are controlled
               // resources for service accounts. (Also used by NextFlow)
-              "roles/iam.serviceAccountUser",
-              "roles/lifesciences.editor",
-              "roles/serviceusage.serviceUsageConsumer")
+              "iam.serviceAccounts.actAs",
+              "iam.serviceAccounts.get",
+              "iam.serviceAccounts.list",
+              "lifesciences.workflows.run",
+              "lifesciences.operations.cancel",
+              "serviceusage.services.use")
           .build();
-  // Currently, workspace editors, applications and owners have the sam cloud permissions as
-  // writers. If that changes, create a new list and modify the map below.
-  public static final ImmutableMap<WsmIamRole, List<String>> CLOUD_SYNC_ROLE_MAP =
+
+  private static final CustomGcpIamRole PROJECT_READER =
+      new CustomGcpIamRole("PROJECT_READER", READER_PERMISSIONS);
+  private static final CustomGcpIamRole PROJECT_WRITER =
+      new CustomGcpIamRole("PROJECT_WRITER", WRITER_PERMISSIONS);
+  // Currently, workspace editors, applications and owners have the same cloud permissions as
+  // writers. If that changes, create a new CustomGcpIamRole and modify the map below.
+  public static final ImmutableMap<WsmIamRole, CustomGcpIamRole> CUSTOM_GCP_PROJECT_IAM_ROLES =
       ImmutableMap.of(
-          // TODO: this should map to OWNER_PERMISSIONS if that's created.
-          WsmIamRole.OWNER, WRITER_PERMISSIONS,
-          // TODO: this should map to APPLICATION_PERMISSIONS if that's created.
-          WsmIamRole.APPLICATION, WRITER_PERMISSIONS,
-          WsmIamRole.WRITER, WRITER_PERMISSIONS,
-          WsmIamRole.READER, READER_PERMISSIONS);
+          // TODO: this should map to PROJECT_OWNER if that's created.
```

Comment:
might be simpler to make a `PROJECT_OWNER` and put this comment there. Same for `PROJECT_APPLICATION`.

Reply:
That's true, but it would also create two extra custom roles in each project which wouldn't be used. I think that would end up being more confusing in the end